Devscord.DiscordFramework/Middlewares/Factories/DiscordServerContextFactory.cs

```diff
@@ -1,21 +1,37 @@
-﻿using Devscord.DiscordFramework.Middlewares.Contexts;
+﻿using Autofac;
+using Devscord.DiscordFramework.Middlewares.Contexts;
+using Devscord.DiscordFramework.Services;
 using Discord;
+using System.Collections.Generic;
 using System.Linq;
 
 namespace Devscord.DiscordFramework.Middlewares.Factories
 {
     internal class DiscordServerContextFactory : IContextFactory<IGuild, DiscordServerContext>
     {
+        private readonly UsersService _usersService;
+        private readonly UsersRolesService _usersRolesService;
+        private readonly UserContextsFactory _userContextsFactory;
+        private readonly ChannelContextFactory _channelContextFactory;
+
+        public DiscordServerContextFactory(IComponentContext context)
+        {
+            this._usersService = context.Resolve<UsersService>();
+            this._usersRolesService = context.Resolve<UsersRolesService>();
+            this._userContextsFactory = context.Resolve<UserContextsFactory>();
+            this._channelContextFactory = context.Resolve<ChannelContextFactory>();
+        }
+
         public DiscordServerContext Create(IGuild restGuild)
         {
-            var userFactory = new UserContextsFactory();
-            var channelFactory = new ChannelContextFactory();
+            var systemChannel = this._channelContextFactory.Create(restGuild.GetSystemChannelAsync().Result);
+            var textChannels = restGuild.GetTextChannelsAsync().Result.Select(x => this._channelContextFactory.Create(x));
 
-            var owner = userFactory.Create(restGuild.GetOwnerAsync().Result);
-            var systemChannel = channelFactory.Create(restGuild.GetSystemChannelAsync().Result);
-            var textChannels = restGuild.GetTextChannelsAsync().Result.Select(x => channelFactory.Create(x));
+            UserContext GetOwner() => this._userContextsFactory.Create(restGuild.GetOwnerAsync().Result);
```

Comment:
czemu z dużej litery?

Reply:
domyślnie VS podpowiadał nazwę z dużej, ale to do ustalenia jakie nazewnictwo przyjmujemy dla lokalnych funkcji